src/common/transformations/include/transformations/common_optimizations/constants_reduce.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "transformations_visibility.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API ConstantsReduce : public ov::pass::GraphRewrite {
```

Comment:
usually we use `ov::pass::GraphRewrite` as a container for matcher passes to execute them efficiently 
it's better to use `ModelPass` instead of `GraphRewrite` if you do not plan to combine several matchers inside this transformation

Reply:
also we can change it to `MatherPass` and match a constant with a condition (predicate)
e.g. predicate = const_node->get_byte_size() > 256

Reply:
https://docs.openvino.ai/2025/documentation/openvino-extensibility/transformation-api/matcher-pass.html